x-pack/plugins/observability_solution/dataset_quality/common/types/common.ts

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+export type SortDirection = 'asc' | 'desc';
+
+export type Maybe<T> = T | null | undefined;
+
+export interface Coordinate {
+  x: number;
+  y: Maybe<number>;
```

Comment:
Will it ever by a nullable as it looks like `timeSeriesBucket.doc_count` will always be a number in:

```
timeSeries: bucket.timeSeries.buckets.map((timeSeriesBucket) => ({
  x: timeSeriesBucket.key,
  y: timeSeriesBucket.doc_count,
})),
```

Reply:
This has been designed like this to support 2 use cases. 
1. Charts can only show broken lines when Y = null.
2. Charts will be continuous but go down drastically when Y = 0

At the moment we are exploring the best way to display the data. We tried null first and then decided to go with 0 at the moment.

But a Y axis in a Coordinate should actually be both 0 or null given how the chart should be rendered, continuous or broken. Hence i gave it both options. This will allow us to define future coordinates as well.